MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.cpp

```diff
@@ -95,7 +95,7 @@ double PiecewiseLinearInterpolation::getDerivative(
         interval_idx = 1;
     }
 
-    if (interval_idx > 2 && interval_idx < _supp_pnts.size() - 1)
+    if (interval_idx > 1 && interval_idx < _supp_pnts.size() - 2)
```

Comment:
Regarding this, I just follow the OGS5 implementation. @wenqing could you please have a check?